contracts/governance/utils/VotesOverridable.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {Checkpoints} from "../../utils/structs/Checkpoints.sol";
+import {Votes} from "./Votes.sol";
+import {SafeCast} from "../../utils/math/SafeCast.sol";
+
+/**
+ * @dev Extension of {Votes} that adds support for checkpointed delegations and balances. This is required
+ * to use the `GovernorOverrideDelegateVote` extension.
+ */
+abstract contract VotesOverridable is Votes {
+    using Checkpoints for Checkpoints.Trace208;
+
+    error VotesOverridableFutureLookup(uint256 timepoint, uint256 currentTimepoint);
```

Comment:
For the futur: maybe we should have this extension be an ERC (that depend on ERC-5805) and use a custom error that is similar to `ERC5805FutureLookup`

Reply:
We might use `ERC5805FutureLookup` here too. It's not officially part of the ERC so it doesn't have a strict definition. I think it's a good fit here